### DISCUSSION TITLE
[Question] Does anyone know how to broadcast raw sign transaction of Polkadot from Wallet-core  to testnet Westend or mainnet Polkadot?

### DISCUSSION THREAD

**Discussion by Raden-Hor** [trustwallet/wallet-core#1990] [Q&A] [answered]

**Comment by gituser** [comment, marked as answer]
In order to sign a testnet transaction you need to patch a wallet-core first for the Westend support as it differs from the mainnet.
For signing the transaction you need to know certain transaction-material which can be obtained from the node.
You can do a JSON or Protobuf signing, see examples in the repository (tests folder).

**Reply by Raden-Hor** [reply]
Thank for your answer, But Do you know where we can push raw transaction to Westend?

**Reply by hewigovens** [reply]
You need to connect to the Westend node and call rpc https://polkadot.js.org/docs/substrate/rpc/#submitextrinsicextrinsic-extrinsic-hash

**Reply by gituser** [reply]
@Raden-Hor all this data to construct a transaction available through `transaction-material` RPC call to sidecar: https://paritytech.github.io/substrate-api-sidecar/dist/

but you need to find one for the Westend and also modify wallet-core as it doesn't support Westend and Westend is different to mainnet.

And then you push your signed transaction through either sidecar's `/transaction` API end or through polkadot's node as @hewigovens mentioned

**Reply by Raden-Hor** [reply]
Thank you for your answer @gituser But do u know how to modify wallet-core to support Westend, There any document?

**Reply by gituser** [reply]
There is not, you need to look into source code and understand how it works.
It's actually very trivial when you get it.
Wallet-core devs are not supporting testnets (though would be nice if they do) as testing on real coins involve potential money loss, so you have to figure out this on your own or pay someone to do the job for you.

**Reply by Raden-Hor** [reply]
Thank u @gituser
